### DISCUSSION TITLE
Using variable negation in switch

### DISCUSSION THREAD

**Discussion by sauramud** [savonet/liquidsoap#2768] [Q&A] [answered]
Hello, I'm wondering how it's possible to do a thing like this:
I need to have a variable like this

`xmaslist = interactive.bool("xmas_playlist", false)`

then in my switch I want simply to have a

```
    switch(id = "primary_play", [
      ({13h30-13h45},  delay(1800., dmix)),
      (xmaslist, xmasmix),
      (!xmaslist,daymix)
    ])
```

So, I've got a var in my script that simply, when I put it to true i Play a xmas list instead of the normal daymix. When xmas time is ended I simply put to false xmaslist then exclude xmasmix and reinclude daymix.
Can anyone help please ?
Thanks

**Comment by smimram** [comment, marked as answer]
The thing here is that `xmaslist` is not a boolean, (which would always have the same value) but a boolead _getter_ (which can change value over time). For this reason, you need to
- get the value (with `getter.get`)
- apply negation (with `!`)
- and make a getter again (with `{...}`).

In this case, this would give something like
```
    switch(id = "primary_play", [
      ({13h30-13h45},  delay(1800., dmix)),
      (xmaslist, xmasmix),
      ({!getter.get(xmaslist)},daymix)
    ])
```
Since this is the last and default case, you could also simply do here
```
    switch(id = "primary_play", [
      ({13h30-13h45},  delay(1800., dmix)),
      (xmaslist, xmasmix),
      ({true},daymix)
    ])
```

**Comment by sauramud** [comment]
Thanks @smimram 
I was missing the getter.get syntax.
The last true is valued and executed only if all previous are negative then?
Thanks!

**Reply by smimram** [reply]
Yes, `switch` evaluates conditions in order and stops at the first one which returns `true`.

BTW, `getter.get` can be replaced by applying `()` ie `getter.get(xmaslist)` is equivalent to `xmaslist()`, but I find it more clear.
